runtime/src/iree/hal/drivers/hip/stream_command_buffer.c

```diff
@@ -478,6 +478,16 @@ static iree_status_t iree_hal_hip_stream_command_buffer_dispatch(
       z0, iree_hal_hip_native_executable_lookup_kernel_params(
               executable, entry_point, &kernel_params));
 
+  // If any of the workgroup counts/block dims are zero, we can skip execution
```

Comment:
the block dims will never be zero (or if they are, that should be checked when loading the executable) - you just need to check the workgroup count and can do it above before flushing collectives